src/parameterized_tendencies/microphysics/precipitation.jl

```diff
@@ -211,6 +211,7 @@ function compute_precipitation_cache!(Y, p, colidx, ::Microphysics1Moment, _)
     α(ts) = TD.Parameters.cv_l(thp) / Lf(ts) * (Tₐ(ts) - cmp.ps.T_freeze)
 
     qₚ(ρqₚ, ρ) = max(FT(0), ρqₚ / ρ)
+    limit(q, dt) = q / dt
```

Comment:
I know this is only rearranging what we had before, so I presume the answer is no, but shouldn't `dt` here be the RK stage `dt`, rather than the full `dt`?

Reply:
It should be RK stage `dt` but the limiting should be done on the full tendency and not individual components of it. 

The problem with that is that if I were to limit the full tendency I would have to know how it affects the individual components. Because I would have to know how to adjust the corresponding source terms to density, energy, etc. That's complicated, because for example some sink terms to rain lead to changes in density/energy of the working fluid, but others do not.

So to keep it simple for now, we instead apply the limiter on each tendency component and use the `dt` and not RK `dt`. It's not accurate, but the goal is to keep large negative numbers from developing, not necessarily to always guarantee positivity. We were even discussing trying to limit by more than just `dt` and testing how that would affect the stability vs accuracy.

Reply:
Ahh ok, thank you for explaining! As long as it's an intentional choice ;-)